astropy/coordinates/baseframe.py

```diff
@@ -761,22 +811,36 @@ def _apply(self, method, *args, **kwargs):
         kwargs : dict
             Any keyword arguments for ``method``.
         """
+        if '_framedata' in kwargs:
+            data = kwargs.pop('_framedata')
+        else:
+            data = self.data if self.has_data else None
+
         def apply_method(value):
             if isinstance(value, ShapedLikeNDArray):
-                return value._apply(method, *args, **kwargs)
+                if method=='replicate' and not hasattr(value, method):
```

Comment:
Actually, it wasn't just future-proofing (although that is indeed an important side-benefit!). It turns out this is necessary just to get the tests to pass because `Quantity` sub-classes like `Angle`/`Latitude`/`Longitude` don't have `replicate`. 

But in the end it's good anyway this way, I think!